Allow PID file path to be relative when daemonize a process (scheduler, kerberos, etc)

XD-DENG opened the pull request:
Closes #13200  (please refer to #13200 for detailed background and discussion.)

Currently, if the PID file path provided is relative, the process silently fails to launch.

This fix ensures the path specified to be a normalized absolutized path eventually (expand with cwd), no matter the given value is relative or absolute.

This PR also tries to increase coverage a little bit.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

ashb commented on airflow/utils/cli.py:
```diff
@@ -227,8 +227,12 @@ def setup_locations(process, pid=None, stdout=None, stderr=None, log=None):
         pid = os.path.join(settings.AIRFLOW_HOME, f'airflow-{process}.pid')
+    else:
+        if not os.path.isabs(pid):
+            raise AirflowException("Path of PID file must be absolute")
```
We should make it relative to cwd instead. https://docs.python.org/3/library/os.path.html#os.path.abspath

ashb replied:
Current working dir is what every unix program treats a relative path as -- we should be no different.

XD-DENG replied:
That's what I thought of and proposed in the beginning (https://github.com/apache/airflow/issues/13200#issuecomment-749128925). Please check the discussion at #13200

XD-DENG replied:
My personal preference aligns with your suggestion.

The current implementation is a "mid-point agreed" after the discussion. Or we should further elaborate the discussion.

potiuk replied:
As far as I see the "https://linux.die.net/man/1/daemonize" command prints `The 'path' parameter must be an absolute path name` error when passed a relative one.  I am not very strong on that, but I think this is expected from a "daemon" program.

XD-DENG committed:
Allow PID file path to be relative when daemonize a process (scheduler, kerberos, etc)

Closes #13200.

Currently, if the PID file path provided is relative,
the process silently fails to launch.

This fix ensures the PID path specified to be
a normalized absolutized path eventually (expand with cwd),
no matter the given value is relative or absolute.
